src/NServiceBus.NHibernate.AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.1.1/Reliability/Outbox/When_headers_keys_contain_special_characters.cs

```diff
@@ -0,0 +1,95 @@
+﻿namespace NServiceBus.AcceptanceTests.Outbox
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NUnit.Framework;
+
+    public class When_headers_contain_special_characters : NServiceBusAcceptanceTest
+    {
+        static Dictionary<string, string> sentHeaders = new Dictionary<string, string>
+                          {
+                            { "a-B1", "a-B" },
+                            { "a-B2", "a-ɤϡ֎ᾣ♥-b" },
+                            { "a-ɤϡ֎ᾣ♥-B3", "a-B" },
+                            { "a-B4", "a-\U0001F60D-b" },
+                            { "a-\U0001F605-B5", "a-B" },
+                            { "a-B6", "a-😍-b" },
+                            { "a-😅-B7", "a-B" },
+                            {"a-b8", "奥曼克"},
+                            {"a-B9", "٩(-̮̮̃-̃)۶ ٩(●̮̮̃•̃)۶ ٩(͡๏̯͡๏)۶ ٩(-̮̮̃•̃)" },
+                            {"a-b10", "தமிழ்" }
+                        };
+
+        [Test]
+        public async Task Outbox_should_work()
+        {
+            var context =
+                await Scenario.Define<Context>()
+                .WithEndpoint<OutboxEndpoint>(b => b.When(session => session.SendLocal(new Pläceörder())))
+                .Done(c => c.MessageReceived)
+                .Run(TimeSpan.FromSeconds(20));
+
+            Assert.IsNotEmpty(context.UnicodeHeaders);
+            CollectionAssert.IsSubsetOf(sentHeaders, context.UnicodeHeaders);
```

Comment:
I switched over to use IsSupersetOf because it gives a more apropriate message on failure (need to be careful with the parameters though)